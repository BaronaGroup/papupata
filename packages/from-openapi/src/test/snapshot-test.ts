import { generatorTest } from './generatorTest'

describe('snapshots', () => {
  it('trivial', generatorTest('./trivial.yaml'))
  it('basic', generatorTest('./basic.yaml'))
  it('body', generatorTest('./with-body.yaml'))
  it('overview', generatorTest('./overview.yaml'))
})
