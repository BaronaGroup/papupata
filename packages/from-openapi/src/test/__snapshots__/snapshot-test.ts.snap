// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots basic 1`] = `
"// Generated using papupata-from-openapi
import { APIDeclaration } from 'papupata'
const declarationBase = new APIDeclaration()
export const generatedAPI = {
  _papupataApiDeclaration: declarationBase,
  /**
   * @summary Returns a list of users.
   * @description Optional extended description in CommonMark or HTML.
   */
  Users: declarationBase.declareGetAPI('/users').response<Array<string>>(),
}
"
`;

exports[`snapshots body 1`] = `
"// Generated using papupata-from-openapi
import { APIDeclaration } from 'papupata'
const declarationBase = new APIDeclaration()
export const generatedAPI = {
  _papupataApiDeclaration: declarationBase,
  /**
   * @summary Returns a list of users.
   * @description Optional extended description in CommonMark or HTML.
   */
  Users: declarationBase.declarePostAPI('/users').body<{ username: string; password: string }>().response<string>(),
}
"
`;

exports[`snapshots overview 1`] = `
"// Generated using papupata-from-openapi
import { APIDeclaration } from 'papupata'
const declarationBase = new APIDeclaration()
export const generatedAPI = {
  _papupataApiDeclaration: declarationBase,
  /**
   * @summary List API versions
   */
  listVersionsv2: declarationBase.declareGetAPI('/').response<unknown>(),

  /**
   * @summary Show API version details
   */
  getVersionDetailsv2: declarationBase.declareGetAPI('/v2').response<unknown | unknown>(),
}
"
`;

exports[`snapshots trivial 1`] = `
"// Generated using papupata-from-openapi
import { APIDeclaration } from 'papupata'
const declarationBase = new APIDeclaration()
export const generatedAPI = { _papupataApiDeclaration: declarationBase }
"
`;
